### PR TITLE
(DOCSP-23224): TOC refactor

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
GDoc with discussion and proposal: https://docs.google.com/document/d/1A1zNsZGr7DLQakYBh-o1gaBlJIJCYz2qAdxVWC2XJjU/edit#

Staging: https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-shell/docsworker-xlarge/DOCSP-23224/

### CODE REVIEW

File: source/includes/fact-mongoshrc-loading.rst [hunk 1]
```diff
@@ -1,24 +1,20 @@
-On startup, :binary:`mongosh` checks the user's ``HOME`` directory for
-a JavaScript file named :ref:`.mongoshrc.js <mongoshrc-js>`. If this
-file is found, :binary:`mongosh` reads the content of
-:ref:`.mongoshrc.js <mongoshrc-js>` before displaying the prompt for
-the first time.
+On startup, :binary:`mongosh` checks your ``HOME`` directory for a
```
Comment (davemungo) [inline source/includes/fact-mongoshrc-loading.rst:1]: Should .mongoshrc.js be enclosed in backticks, ``.mongoshrc.js`` ?

File: source/includes/fact-mongoshrc-loading.rst [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
 
-   The legacy :binary:`~bin.mongo` shell used a configuation file
-   called ``.mongorc.js``. If :binary:`mongosh` detects this file on
-   startup, and :ref:`.mongoshrc.js <mongoshrc-js>` is not present,
-   :binary:`mongosh` will suggest renaming ``.mongorc.js`` to
-   :ref:`.mongoshrc.js <mongoshrc-js>`. 
-
-   The legacy ``.mongorc.js`` file is not loaded.
-
+   The legacy :binary:`~bin.mongo` shell used a configuation file called
```
Comment (davemungo) [inline source/includes/fact-mongoshrc-loading.rst:16]: The formatting for ``.mongorc.js`` and ``.mongoshrc.js`` should be the same

File: source/mongoshrc.txt [hunk 3]
```diff
@@ -18,35 +18,6 @@
...
-.. include:: /includes/examples/ex-display-dbname-and-hostname.rst
-
-.. _ex-display-system-up-time:
-
-Display System Up Time and Document Count
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. include:: /includes/examples/ex-display-system-uptime.rst
-
+You can use the ``.mongoshrc`` file to customize ``mongosh`` startup
```
Comment (davemungo) [inline source/mongoshrc.txt:21]: I don't see the ref where these examples are linked. Could it have been dropped by mistake?

**Review by davemungo — COMMENTED** [review]
Thanks @jeff-allen-mongo  looks good, just a couple fo small comments

File: source/includes/fact-mongoshrc-loading.rst (hunk 1 above)
Comment (jeff-allen-mongo) [reply source/includes/fact-mongoshrc-loading.rst:1]: It was like this originally, but this page looked really bad with the monospace styling for mongoshrc.js. The frequency at which we use the term made the page really hard to read with constant switching of styles, so I opted for plaintext.

File: source/includes/fact-mongoshrc-loading.rst (hunk 2 above)
Comment (jeff-allen-mongo) [reply source/includes/fact-mongoshrc-loading.rst:16]: Fixed

File: source/mongoshrc.txt (hunk 3 above)
Comment (jeff-allen-mongo) [reply source/mongoshrc.txt:24]: I intentionally removed the examples from this page to convert it to strictly a reference page specific to the .mongoshrc file. The examples are covered in these two places:

- https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-shell/docsworker-xlarge/DOCSP-23224/write-scripts/#std-label-mongosh-write-scripts-config-file
- https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-shell/docsworker-xlarge/DOCSP-23224/reference/customize-prompt/

There is a link to the Write Scripts section on the reference page. I'll add a heading so it's a bit more obvious.

File: source/includes/fact-mongoshrc-loading.rst (hunk 2 above)
Comment (jeff-allen-mongo) [reply source/includes/fact-mongoshrc-loading.rst]: https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-shell/docsworker-xlarge/DOCSP-23224/mongoshrc/

**Comment by jeff-allen-mongo** [comment]
@davemungo thanks for the review. I've updated and responded to your comments.

File: source/includes/fact-mongoshrc-loading.rst (hunk 1 above)
Comment (davemungo) [reply source/includes/fact-mongoshrc-loading.rst:1]: FWIW - Our styling is way too busy, I wish we had more say in how the docs look.

File: source/mongoshrc.txt [hunk 4]
```diff
@@ -18,35 +18,9 @@
...
-Display System Up Time and Document Count
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. include:: /includes/examples/ex-display-system-uptime.rst
+Example Usage
+-------------
 
+You can use the .mongoshrc file to customize ``mongosh`` startup
+behavior and add scripted functionality. To see examples, see
+:ref:`mongosh-write-scripts-config-file`.
```
Comment (davemungo) [inline source/mongoshrc.txt:26]: Thanks - I found the ref anchor.

GH didn't show the file diff by default so ctrl+F couldn't find the other mongosh-write-scripts-config-file on this page.

**Review by davemungo — APPROVED** [review]
Thanks @jeff-allen-mongo  LGTM

File: source/configure-mongosh.txt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Read the following pages to learn how to configure ``mongosh`` behaviors
```
Comment (mdb-ashley) [inline source/configure-mongosh.txt:15]: [Suggestion]

You could consider editing this down to "Learn how to configure ``mongosh`` behaviors and user experience:".

File: source/includes/examples/ex-display-line-numbers.rst [hunk 6]
```diff
@@ -1,5 +1,5 @@
-This code will dynamically update the :binary:`~bin.mongosh` prompt
-to display line numbers:
+To update the :binary:`~bin.mongosh` prompt to display line numbers, run
```
Comment (mdb-ashley) [inline source/includes/examples/ex-display-line-numbers.rst:1]: [suggestion]

Consider, "To display line numbers in the :binary:`~bin.mongosh` prompt, run..."

File: source/includes/fact-mongoshrc-loading.rst [hunk 7]
```diff
@@ -1,24 +1,20 @@
...
-   startup, and :ref:`.mongoshrc.js <mongoshrc-js>` is not present,
-   :binary:`mongosh` will suggest renaming ``.mongorc.js`` to
-   :ref:`.mongoshrc.js <mongoshrc-js>`. 
-
-   The legacy ``.mongorc.js`` file is not loaded.
-
+   The legacy :binary:`~bin.mongo` shell used a configuration file called
+   .mongorc.js. If :binary:`mongosh` detects this file on startup,
+   and .mongoshrc.js is not present, :binary:`mongosh` does not load
+   the legacy .mongorc.js file, and will suggest renaming
```
Comment (mdb-ashley) [inline source/includes/fact-mongoshrc-loading.rst:19]: [suggestion]

Consider "...and suggests renaming..."

File: source/reference/configure-shell-settings-api.txt [hunk 8]
```diff
@@ -1,8 +1,8 @@
 .. _configure-settings-api:
 
-==============================================
-Configure :binary:`~bin.mongosh` Using the API
-==============================================
+================================
+Configure Settings Using the API
```
Comment (mdb-ashley) [inline source/reference/configure-shell-settings-api.txt:4]: [praise]

I really like this title change. Nice work.

File: source/reference/editor-mode.txt [hunk 9]
```diff
@@ -1,8 +1,8 @@
 .. _mongosh-editor-mode:
 
-===========
-Editor Mode
-===========
+==========================
+Use an Editor for Commands
```
Comment (mdb-ashley) [inline source/reference/editor-mode.txt:4]: [praise]

This title change makes everything clearer for the user. Excellent choice.

**Review by mdb-ashley — APPROVED** [review]
LGTM with a few small suggestions. Thanks for handling this, Jeff!

**Review by gribnoysup — APPROVED** [review]
Looks good! I really like the new TOC order, makes a lot of sense to me 👍

**Pull request merged**